### ISSUE TITLE
var fsize = function (cb) {}

### ISSUE THREAD

**Issue by oeuillot** [leetreveil/musicmetadata#93]
Hello,

In the function  (index.js:16) 
`````
var fsize = function (cb) {
    if (stream.hasOwnProperty('path')) {
...
    } else if (stream.hasOwnProperty('fileSize')) {
....
    } else if (opts.fileSize) {
````
If the user specify the fileSize explicitly in the opts object :
Is-it possible to get the opts.fileSize first ?

Regards,

**Comment by leetreveil** [comment]
Ah ok so you want to override the filesize calculation done in the `if (stream.hasOwnProperty('path'))` branch?

I can't see any issues with putting `if (opts.fileSize) {` to the top if you want to open a pull request.

**Comment by oeuillot** [comment]
Ok thanks, I will propose you a pull request soon :-)

Regards,

**Comment by kf6kjg** [comment]
Sorry oeuillot, beat you to it: https://github.com/leetreveil/musicmetadata/pull/94

I needed that fix myself, and I also added initial support for request.js - something I also was needing for my project.  Couldn't easily keep both changes separate.

**Comment by leetreveil** [comment]
Fixed in b33bcca4239f8c849a4a8db6bc60ea0bad1131af
